internal/reminder/check.go

```diff
@@ -0,0 +1,70 @@
+package reminder
+
+import (
+	"context"
+	"hellper/internal/bot"
+	"hellper/internal/log"
+	"hellper/internal/model"
+)
+
+type notifyRules struct {
+	snoozedUntil bool
+	lastPin      bool
+	slaClose     bool
+	status       string
+}
+
+// CanSendNotify checks the notification rules
+func CanSendNotify(ctx context.Context, client bot.Client, logger log.Logger, repository model.Repository, incident model.Incident) bool {
+	logger.Info(
+		ctx,
+		log.Trace(),
+		log.Action("running"),
+		log.NewValue("channelID", incident.ChannelId),
+		log.NewValue("channelName", incident.ChannelName),
+	)
+
+	rules := notifyRules{
+		snoozedUntil: hasSnoozedUntil(ctx, logger, incident),
```

Comment:
Should we check this rule before any check related with status? Because if this is true it should return false, no matter what.

Reply:
I thought about that, but if I do that, in the future we will not be able to behave differently. For example, do not verify it in the resolved status. The way I coded it was thinking about the possibility of exclusive testing for each status.

Reply:
Makes sense, the rules are separated in status, agreed!